src/feedback-rating/stars-container.tsx

```diff
@@ -0,0 +1,53 @@
+import { StarContainerData } from "./stars-container-data";
+import { Button, Img, Ul } from "./stars-container.styles";
+import { StarsContainerProp } from "./types";
+import StarFilledIcon from "./star-filled-icon.png";
+import StarUnfilledIcon from "./star-unfilled-icon.png";
+
+export const StarsContainer = ({
+    ratings,
+    onRatingsChange,
+}: StarsContainerProp): JSX.Element => {
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleStarSelection = (starIndex: number) => {
+        onRatingsChange(starIndex);
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderStar = (starIndex: number) => {
+        const altText = `${starIndex} star${starIndex === 1 ? "" : "s"}${
+            ratings === starIndex ? ", selected" : ""
+        }`;
+        if (starIndex <= ratings) {
+            return <Img src={StarFilledIcon} alt={altText} />;
+        } else {
+            return <Img src={StarUnfilledIcon} alt={altText} />;
+        }
+    };
+
+    const renderRatings = () => {
+        return [...Array(StarContainerData.MAX_STAR)].map((_star, index) => {
+            return (
+                <li key={index + 1}>
+                    <Button
```

Comment:
with a screenreader the results might be confusing as it reads out all the buttons individually: "1 star selected", "2 stars selected", "3 stars unselected" etc.

the rating input can be implemented with a single `input` instead - I have seen examples using `type=radio` and `type=range`:

https://www.w3.org/WAI/tutorials/forms/custom-controls/
https://dev.to/madsstoumann/star-rating-using-a-single-input-i0l

Reply:
the input range implementation above is still not ideal because it requires very specific clicking and doesn't provide visual feedback when you hover over a specific star

Reply:
updated to use the `type=radio` implementation